src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java

```diff
@@ -39,22 +42,43 @@ public static TripMonitorNotification createAlertNotification(
         return notification;
     }
 
+    /**
+     * Create a new notification about a change in the trip's arrival or departure time exceeding a threshold.
+     *
+     * @param delayInMinutes The delay in minutes (negative values indicate early times).
+     * @param delayThresholdMinutes The user's threshold for delay notifications in minutes.
+     * @param targetDatetime The actual arrival or departure of the trip
+     * @param delayType Whether the notification is for an arrival or departure delay
+     */
     public static TripMonitorNotification createDelayNotification(
         long delayInMinutes,
-        int delayThreshold,
-        NotificationType delayType)
-    {
+        int delayThresholdMinutes,
+        Date targetDatetime,
+        NotificationType delayType
+    ) {
         TripMonitorNotification notification = new TripMonitorNotification();
         notification.type = delayType;
         if (delayType != NotificationType.ARRIVAL_DELAY && delayType != NotificationType.DEPARTURE_DELAY) {
             LOG.error("Delay notification not permitted for type {}", delayType);
             return null;
         }
+        String delayHumanTime;
+        if (Math.abs(delayInMinutes) <= 1) {
+            delayHumanTime = "about on time";
+        } else if (delayInMinutes > 0) {
+            delayHumanTime = String.format("%d minute%s late", delayInMinutes, delayInMinutes > 1 ? "s" : "");
+        } else {
+            delayHumanTime = String.format("%d minute%s early", delayInMinutes, delayInMinutes < -1 ? "s" : "");
+        }
+
         notification.body = String.format(
-            "The %s time for your itinerary was delayed by %d minutes (your threshold is currently set to %d minutes).",
+            "The %s time for your itinerary is now %s (%s) (your threshold is currently set to %d minutes).",
             delayType == NotificationType.ARRIVAL_DELAY ? "arrival" : "departure",
-            delayInMinutes,
-            delayThreshold
+            ZonedDateTime
+                .ofInstant(targetDatetime.toInstant(), DateTimeUtils.getOtpZoneId())
+                .format(DateTimeFormatter.ofPattern("HH:mm")),
```

Comment:
This time format should be read from a config value. Perhaps it could be called `NOTIFICATION_TIME_FORMAT`?

Reply:
Fixed in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14